### DISCUSSION TITLE
Enabling Network Switching in Web3Modal v3 on User-Defined Events

### DISCUSSION THREAD

**Discussion by amandeep-coolplugin** [WalletConnect/walletconnect-monorepo#3710] [Developer Support]
Hi there

Is there a hook in Web3Modal version 3 that allows for network switching in response to a specific event? For instance, I'd like to switch the network within the wallet when the user selects a predefined network.
using below docks code.
https://docs.walletconnect.com/2.0/web3modal/v3/react/about

**Comment by glitch-txs** [comment]
Hi, Web3Modal follows [CAIP-25](https://chainagnostic.org/CAIPs/caip-25). The chains are proposed in the sessions so the user won't need to approve them on the wallet side in order for you to change them in the dapp. Changing a chain in this context means to set a default chain to target the transaction the user will trigger to.
Adding to this Web3Modal supports all Wagmi's hooks, including [useSwitchNetwork](https://wagmi.sh/react/hooks/useSwitchNetwork) one.

**Reply by amandeep-coolplugin** [reply]
Hi,
Thank you for your response; I was able to achieve my requirement using the method you mentioned. However, I have another question: are there any settings in Web3Modal to display only specific wallets during the connection process, as shown in the image in this example?https://prnt.sc/-smYhhKQjHT2
I have tried using custom connectors settings but still other wallets are shown during the connections.
https://web3modal.com/

**Reply by glitch-txs** [reply]
Hi, yes, you can replace the ones you see there with featuredWalletIds option
https://docs.walletconnect.com/web3modal/react/options#featuredwalletids

**Reply by amandeep-coolplugin** [reply]
No, you get me wrong , i want to show only walletconnect and metamask wallet option in modal.
Like below image.https://prnt.sc/ohz6EojmAzYd

**Reply by glitch-txs** [reply]
It's not possible at the moment but it's going to be added in the next release! Stay tuned to [Web3Modal Repo](https://github.com/WalletConnect/web3modal) :)
